Code/Tools/ProjectManager/Source/CreateAGemScreen.h

```diff
@@ -34,36 +38,54 @@ namespace O3DE::ProjectManager
         explicit CreateGem(QWidget* parent = nullptr);
         ~CreateGem() = default;
 
+        void ClearFields();
+
+        ProjectManagerScreen GetScreenEnum() override
+        {
+            return ProjectManagerScreen::CreateGem;
+        }
+
     signals:
         void GemCreated(const GemInfo& gemInfo);
 
-    private slots:
+    protected slots:
         void HandleBackButton();
         void HandleNextButton();
+        
+
+    private slots:
         void HandleGemTemplateSelectionTab();
         void HandleGemDetailsTab();
         void HandleGemCreatorDetailsTab();
 
-    private:
-        void LoadButtonsFromGemTemplatePaths(QVBoxLayout* gemSetupLayout);
-        QScrollArea* CreateGemSetupScrollArea();
-        QScrollArea* CreateGemDetailsScrollArea();
-        QScrollArea* CreateGemCreatorScrollArea();
-        QFrame* CreateTabButtonsFrame();
-        QFrame* CreateTabPaneFrame();
+    protected:
         bool ValidateGemTemplateLocation();
         bool ValidateGemDisplayName();
         bool ValidateGemName();
         bool ValidateGemPath();
         bool ValidateFormNotEmpty(FormLineEditWidget* form);
         bool ValidateRepositoryURL();
+        
+        void ProceedToGemDetailsPage();
+        void ProceedToGemCreatorDetailsPage();
+        void ProceedToGemAction();
+
+        virtual void GemAction();
+
+        virtual bool ValidateGemLocation(QDir chosenGemLocation)
```

Comment:
It'd be better to not pass QDir by value, but instead by const reference or pointer.  By doing this and moving the implementation to the .cpp you could move the QDir header include to the .cpp file also and just forward declare it here.  It's good to get in the habit of minimizing header includes which will help keep compile times down.

Also, this function can be const.